specification/dimensions/pricingmodel.md

```diff
@@ -0,0 +1,39 @@
+# Pricing Model
+
+Pricing Model indicates the type of unit price applied to a charge at the time it was used or purchased. Pricing Model is commonly used to differentiate between charges at on-demand vs. reduced prices when investigating optimization opportunities, like for commitment-based discount coverage.
```

Comment:
Should `tiers` be a part of this concept/definition or do we think that's better covered elsewhere?

Reply:
Are you talking about tiered/graduated pricing?

Reply:
Yep, one I've seen recently is IBM Cloud's implementation -- https://cloud.ibm.com/docs/billing-usage?topic=billing-usage-charges

Reply:
I love this! Is that in their cost dataset? Do you know what they call the column? I can't tell if this is a logical explanation or the definition of a value in their data.

There are a few interesting things here:
1. They use `Fixed` for purchases or monthly fees. That matches our intent. Might be good to borrow from their definition if it adds any additional clarity to what we have, if we agree the differentiation is necessary here. (I'm still on the fence given how close it is with On-Demand. But I want to know what others think.)
2. They use `Metered` instead of `On-Demand`. Logically, I can understand why they have that. I personally prefer On-Demand because of it is more explicit about the fact that there aren't any reduced prices in play. But I'm good with whatever the group thinks.
3. Technically, `Tiered` is still an on-demand rate. It's just that they decided to offer lower rates based on how much you use. From that perspective, do we expect practitioners to filter both values in when they want to see everything that wasn't a reduced, on-demand rate? That's the main value of `On-Demand` here. This might be a good, simple poll for the member group.
4. They use `Reserved`. We could use `Committed` if we want to align to the past-tense form for all words. But we do need to explicitly say this is a commitment-based discount. Should we use `Committed Usage` / `Committed Spend`? This should probably align with how we handle `Tiered`, since both splits are provided to share granularity on the levels of savings (Dynamic > Committed Usage > Committed Spend > Tiered > Metered/On-Demand). (Although "Dynamic" (interruptible/spot) is the only one that isn't past-tense 😣)

All thought-provoking things 🤔 Would love to hear others' thoughts on this!

Reply:
I'm still struggling with this (after this many months). I feel like there's two different things we might be mixing up here. One is how the usage/purchase is priced - e.g. Fixed, Metered: Fixed, Metered:Dynamic, Metered:Tiered (seem like the types of values though it could also just be Fixed and Metered and Metered could have a sub categorization for Fixed, Dynamic, and Tiered)
The second is how I ended up paying for it on a particular line - on-demand/full-price OR committed/discounted.

Not sure if that makes sense.. but if we try to merge these two concepts together, it feels like we will have many overlaps.

Reply:
I looked at this at one point and realized that, even if they did overlap, you're ultimately getting one benefit the overrides the rest. If we look at cost, `On-Demand` > `Tiered` > `Dynamic` (spot) > `Committed Spend` > `Committed Usage`. Even if we had tiered spot usage that was covered by commitments, there's ultimately something that's driving the cost for that line item. So if a spend commitment is responsible for zero-ing out the charge, then that's what the Pricing Model value should be set to.

I ran across [this article](https://www.exoscale.com/syslog/cloud-pricing-models/) which covers it as well. I wonder if this is just standard operating procedure for cloud, where it may not make sense to go against the grain 🤔 The more I look at this, the more important I think it is that we differentiate each potential layer of price reduction, like IBM did. I'm a fan of splitting these out to be more detailed.

I propose we keep this as defined right now so we can close the PR, then start a poll to see what everyone thinks about having each of these either together or separate:
1. `Fixed` and `On-Demand`
2. `On-Demand` and `Tiered`
3. `Committed Usage` and `Committed Spend`

Based on the outcome of that, we can either update the closed spec or leave it as-is. Unless anyone sees a major issue, of course. If so, let's get that fixed now.

Reply:
We discussed this in VS2 and decided to keep Pricing Model focused on quantifying the explicit actions you take to generate savings. With that tiered pricing wouldn't fit. The good news is we've talked about a separate column (`SKU Tier`) that would identify the pricing tier. That can be used to identify the tiered pricing in a way that will actually convey the level of savings as well. We also have `Commitment Discount Type/Category` to differentiate committed usage and spend, so splitting that here may not be necessary.